src/overlays/actors/ovl_Obj_Lift/z_obj_lift.c

```diff
@@ -28,46 +38,208 @@ const ActorInit Obj_Lift_InitVars = {
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Lift/func_8093D7A0.s")
+Vec2s D_8093DD60[] = { { 120, -120 }, { 120, 0 },     { 120, 120 }, { 0, -120 },  { 0, 0 },
+                       { 0, 120 },    { -120, -120 }, { -120, 0 },  { -120, 120 } };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Lift/func_8093D88C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Lift/func_8093D8B4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Lift/func_8093D9C0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Lift/func_8093DA48.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Lift/func_8093DB70.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Lift/func_8093DB90.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Lift/ObjLift_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Lift/ObjLift_Draw.s")
+static InitChainEntry sInitChain[] = {
+    ICHAIN_F32_DIV1000(gravity, -600, ICHAIN_CONTINUE),   ICHAIN_F32_DIV1000(terminalVelocity, -15000, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneForward, 4000, ICHAIN_CONTINUE), ICHAIN_F32(uncullZoneScale, 350, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneDownward, 350, ICHAIN_STOP),
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Lift/func_8093DC90.s")
+static f32 D_8093DD98[] = { 0.1f, 0.05f };
+
+static f32 yOffset[] = { -18.0f, -9.0f };
+
+void func_8093D3C0(ObjLift* this, PlayState* play) {
+    Vec3f pos;
+    Vec3f vel;
+    Vec3f* actorPos = &this->dyna.actor.world.pos;
+    s32 i;
+    s32 rnd;
+    s32 pad;
+
+    for (i = 0; i < ARRAY_COUNT(D_8093DD60); i++) {
+        pos.x = (D_8093DD60[i].x * this->dyna.actor.scale.x) + actorPos->x;
+        pos.y = actorPos->y;
+        pos.z = (D_8093DD60[i].z * this->dyna.actor.scale.z) + actorPos->z;
+
+        vel.x = D_8093DD60[i].x * this->dyna.actor.scale.x * 0.8f;
+        vel.y = (Rand_ZeroOne() * 10.0f) + 6.0f;
+        vel.z = D_8093DD60[i].z * this->dyna.actor.scale.z * 0.8f;
+
+        if ((s32)Rand_Next() > 0) {
+            rnd = 64;
+        } else {
+            rnd = 32;
+        }
+
+        EffectSsKakera_Spawn(play, &pos, &vel, actorPos, -0x100, rnd, 15, 15, 0,
+                             ((Rand_ZeroOne() * 50.0f) + 50.0f) * this->dyna.actor.scale.x, 0, 32, 50, -1,
+                             OBJECT_D_LIFT, gDampeGraveBrownElevatorDL);
+    }
+    if (OBJLIFT_GET_1(&this->dyna.actor) == 0) {
+        func_800BBFB0(play, &this->dyna.actor.world.pos, 120.0f, 12, 120, 100, 1);
+    } else if (OBJLIFT_GET_1(&this->dyna.actor) == 1) {
+        func_800BBFB0(play, &this->dyna.actor.world.pos, 60.0f, 8, 60, 100, 1);
+    }
+}
+
+void ObjLift_Init(Actor* thisx, PlayState* play) {
+    f32 temp_fv0;
+    ObjLift* this = THIS;
+
+    Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
+    temp_fv0 = D_8093DD98[OBJLIFT_GET_1(&this->dyna.actor)];
+    this->dyna.actor.scale.z = temp_fv0;
+    this->dyna.actor.scale.x = temp_fv0;
```

Comment:
Could be a chained assignment?

Reply:
The thing with these chain assignments is that the order usually matters, i tried a few combinations but it didn't work.

Reply:
```suggestion
    this->dyna.actor.scale.x = this->dyna.actor.scale.z = D_8093DD98[OBJLIFT_GET_1(&this->dyna.actor)];
```
This works.